classes/submission/Representation.inc.php

```diff
@@ -85,6 +85,33 @@ function getSubmissionId() {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Get submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a galley from submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId($contextId = null) {
```

Comment:
Yes, there's slight duplication of data, in that the current revision is implicitly the revision data with the highest revision number present in the settings table. I started writing my comment above to determine that using some clever joins, but that started to look like a nightmare for the query cardinalities.

Another option worth considering: what about just using subqueries? OTOH we now require MySQL and PostgreSQL versions that both support them. Including this in a join would look like...


    SELECT ... FROM ... WHERE xyz.revision = (SELECT MAX(revision) FROM submission_settings WHERE submission_id=1);

I do think it's worth a try if we can keep the SQL from getting insane.